waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -254,17 +257,47 @@ private boolean isTableAllowed(String databasePrefix, String database, String ta
     return tblAllowList.contains(table);
   }
 
+  /**
+   * This run in parallel because includeInResults could potentially be slow (wait/retries) for certain slow responding
+   * metastores. Using ExecutorService + Futures to maintain the order. Order is important for example when doing calls
+   * like show databases, we return that grouped/ordered per metastore.
+   */
   @Override
   public List<DatabaseMapping> getDatabaseMappings() {
+    // TODO PD refactor/add same logic for StaticDatabaseMappingService.
     Builder<DatabaseMapping> builder = ImmutableList.builder();
-    synchronized (mappingsByPrefix) {
-      for (DatabaseMapping databaseMapping : mappingsByPrefix.values()) {
-        if (includeInResults(databaseMapping)) {
-          builder.add(databaseMapping);
+    ForkJoinPool customThreadPool = new ForkJoinPool(mappingsByPrefix.size());
```

Comment:
slight concern about the creating and shutting this every time might need to reuse an instance of the ForkJoinPool. Will run some tests to see how fast that is.

Reply:
Ran some tests and creating and shutting down 150 ForkJoinPools takes 2 ms so there seems to be very little overhead just object creation.